src/backend/distributed/metadata/node_metadata.c

```diff
@@ -757,6 +759,24 @@ master_update_node(PG_FUNCTION_ARGS)
 		if (force)
 		{
 			handle = StartLockAcquireHelperBackgroundWorker(MyProcPid, lock_cooldown);
+			if (!handle)
+			{
+				/*
+				 * We failed to start a background worker, which probably means that we exceeded
+				 * max_worker_processes, and this is unlikely to be resolved by retrying. We do not want
+				 * to repeatedly throw an error because if master_update_node is called to complete a
+				 * failover then finishing is the only way to bring the cluster back up. Therefore we
+				 * give up on killing other backends and simply wait for the lock. We do set
+				 * lock_timeout to lock_cooldown, because we don't want to wait forever to get a lock.
+				 */
+				SetLockTimeoutLocally(lock_cooldown);
+				ereport(WARNING, (errmsg(
+									  "could not start background worker to kill backends with conflicting"
+									  " locks to force the update. Degrading to acquiring locks "
+									  "with a lock time out."),
+								  errhint(
+									  "Increasing max_worker_processes might help.")));
```

Comment:
very minor: could use a better alignment for that:

```suggestion
				ereport(WARNING, (errmsg("could not start background worker to kill "
										 "backends with conflicting locks to force "
										 "the update. Degrading to acquiring locks "
										 "with a lock time out."),
								  errhint("Increasing max_worker_processes might help.")));
```